### PR TITLE
Added support for filling rounded rectangles

### PR DESCRIPTION
**Pull request by mikeoliphant**
## Description
This PR adds support for filling rounded rectangles.

## Motivation and Context
Currently, any fill options passed to the graphics DrawRectangle calls will be ignored if the rectangle has rounded corners. This PR adds support for the fill options - including opacity and gradient fills.

## How Has This Been Tested?
Tested locally on M5Core2.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

### CODE REVIEW

File: src/nanoFramework.Graphics/Graphics/Core/GraphicsDriver.cpp [hunk 1]
```diff
@@ -290,6 +288,35 @@ void GraphicsDriver::DrawRectangleNative(
...
+
+void GraphicsDriver::FillRectangleNative(const PAL_GFX_Bitmap &bitmap, GFX_Brush &brush, const GFX_Rect &rectangle)
+{
+    int x = rectangle.left;
+    int y = rectangle.top;
+    int width = rectangle.Width();
+    int height = rectangle.Height();
+
+    int xSrc = 0, ySrc = 0;
+    if (ClipToVisible(bitmap, x, y, width, height, NULL, xSrc, ySrc) == false)
```
Comment (josesimoes) [inline src/nanoFramework.Graphics/Graphics/Core/GraphicsDriver.cpp:313]: open and close brackets please

**Review by josesimoes — COMMENTED** [review]
Impressive! 😯 👏🏻 👏🏻 👏🏻

File: src/nanoFramework.Graphics/Graphics/Core/GraphicsDriver.cpp (hunk 1 above)
Comment (mikeoliphant) [reply src/nanoFramework.Graphics/Graphics/Core/GraphicsDriver.cpp:313]: Fixed. Is that something that could be added to the nfbot code style check?

Comment (josesimoes) [reply src/nanoFramework.Graphics/Graphics/Core/GraphicsDriver.cpp:313]: Unfortunately that rule isn't available...

**Review by josesimoes — APPROVED** [review]
Thanks! 👍🚀

**Pull request merged**